homeassistant/components/media_player/firetv.py

```diff
@@ -4,166 +4,126 @@
-        """Get the device state. An exception means UNKNOWN state."""
-        try:
-            response = requests.get(
-                DEVICE_STATE_URL.format(
-                    self.proto, self.host, self.port, self.device_id
-                    ), timeout=10).json()
-            return response.get('state', STATE_UNKNOWN)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve device state for %s", self.device_id)
-            return STATE_UNKNOWN
-
-    @property
-    def current_app(self):
-        """Return the current app."""
-        try:
-            response = requests.get(
-                DEVICE_APPS_URL.format(
-                    self.proto, self.host, self.port, self.device_id, 'current'
-                    ), timeout=10).json()
-            _current_app = response.get('current_app')
-            if _current_app:
-                return _current_app.get('package')
-
-            return None
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve current app for %s", self.device_id)
-            return None
-
-    @property
-    def running_apps(self):
-        """Return a list of running apps."""
-        try:
-            response = requests.get(
-                DEVICE_APPS_URL.format(
-                    self.proto, self.host, self.port, self.device_id, 'running'
-                    ), timeout=10).json()
-            return response.get('running_apps')
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve running apps for %s", self.device_id)
-            return None
-
-    def action(self, action_id):
-        """Perform an action on the device."""
-        try:
-            requests.get(DEVICE_ACTION_URL.format(
-                self.proto, self.host, self.port, self.device_id, action_id
-                ), timeout=10)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Action request for %s was not accepted for device %s",
-                action_id, self.device_id)
-
-    def start_app(self, app_name):
-        """Start an app."""
-        try:
-            requests.get(DEVICE_APPS_URL.format(
-                self.proto, self.host, self.port, self.device_id,
-                app_name + '/start'), timeout=10)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not start %s on %s", app_name, self.device_id)
+    adbkey = config.get(CONF_ADBKEY)
+    get_source = config.get(CONF_GET_SOURCE)
+    get_sources = config.get(CONF_GET_SOURCES)
+
+    device = FireTVDevice(host, name, adbkey, get_source, get_sources)
+    adb_log = " using adbkey='{0}'".format(adbkey) if adbkey else ""
+    if not device._firetv._adb:
+        _LOGGER.warning("Could not connect to Fire TV at %s%s", host, adb_log)
+
+        # Debugging
```

Comment:
Debugging?

Reply:
Maybe a better description would be "Configuration troubleshooting." A number of people have had difficulty with the `adbkey` portion of the configuration, including incorrect paths to permission errors. These error messages are to help them to sort out those issues.